packages/web3-eth-contract/src/contract.ts

```diff
@@ -300,7 +371,13 @@ export class Contract<Abi extends ContractAbi>
 
 	private async _contractMethodEstimateGas<
 		Options extends PayableCallOptions | NonPayableCallOptions,
-	>(abi: AbiFunctionFragment, params: unknown[], options?: Options, block?: BlockNumberOrTag) {
+	>(
```

Comment:
same for this `eth_estimateGas`, this function should utilize eth package. Could you add this as well in new issue

Reply:
https://github.com/ChainSafe/web3.js/issues/4754